File: lista-de-exercicios/exercicio-9/NovoFibonacci.ts
export class NovoFibonacci {
    private numero: number;

    public constructor (numero: number) {
        this.numero = numero;
    }

    public contar(): Array<number> {
        let serie: Array<number> = [];
        let f0 = 0;
        let f1 = 1;
        let f = 0;
        
        if (Number.isInteger(this.numero) && this.numero >= 0) {
            serie.push(f0)
            if (this.numero > 0) {
                serie.push(f1)
                if (this.numero > 1) {
                    while (f <= this.numero) {
                        f = f0 + f1;
                        if (f <= this.numero) {
                            serie.push(f);
                            f0 = f1;
                            f1 = f;
                        }
                    }
                } 
            } 
        } else
            throw Error ('Número inválido');
        return serie;
    }
}

File: lista-de-exercicios-java/src/main/java/org/example/exercicio09/NovoFibonacci.java
package org.example.exercicio09;

import java.util.ArrayList;

public class NovoFibonacci {

    public ArrayList<Integer> serializar(int numero) {
        ArrayList<Integer> serie = new ArrayList<>();
        int f0 = 0;
        int f1 = 1;

        validarNumero(numero);

        int i = 0;
        do {
            serie.add(f0);
            int f = f0 + f1;
            f0 = f1;
            f1 = f;
            i++;
        } while (f0 <= numero);

        return serie;
    }

    private void validarNumero(int numero) {
        if (numero < 0) {
            throw new IllegalArgumentException("Número inválido");
        }
    }
}